### PR TITLE
DMP-2404 Update security data: Super User & Controller Authorisation

### PR DESCRIPTION
**Pull request by richardbull**
### JIRA link (if applicable) ###

[DMP-2404](https://tools.hmcts.net/jira/browse/DMP-2404)

### Change description ###

Add SUPER_USER to globalAccessSecurityRoles controller endpoints (except annotations and createUser)

Audio
* GET /audio/hearings/{hearing_id}/audios - Media metadata for provided hearing
* GET /audio/preview/{media_id} - Preview audio

Audio Requests
* POST /audio-requests - Users can request audio for a hearing and date/time periods
* DELETE /audio-requests/transformed_media/{transformed_media_id} - Delete Transformed Media and Processed Audio
* PATCH /audio-requests/transformed_media/{transformed_media_id} - Update last accessed timestamp for transformed_media
* DELETE /audio-requests/{media_request_id} - Delete Audio Request and Processed Audio
* GET /audio-requests/download - Download audio
* GET /audio-requests/playback - Playback audio

Cases
* GET /cases/{case_id} - Allows the retrieval of a single case using a given case_id
* GET /cases/{case_id}/hearings - Allows the retrieval of hearings for a given case_id
* GET /cases/{case_id}/transcripts - Allows the retrieval of transcripts for a given case_id

Courthouses
* GET /admin/regions - Gets all regions
* GET /admin/courthouses/{courthouse_id} - Get a courthouse record with specified id.

Hearings
* GET /hearings/{hearing_id} - Allows the retrieval of a hearing by ID.
* GET /hearings/{hearing_id}/events - Allows the retrieval of events for a given hearing_id
* GET /hearings/{hearing_id}/transcripts - Allows the retrieval of transcripts for a given hearing_id

Retention
* GET /retentions - Get retention history
* POST /retentions - allow case retention changes to be applied.

Transcription
* POST /transcriptions - Requesters can request transcription
* GET /transcriptions/{transcription_id} - Gets details of a transcription
* PATCH /transcriptions/{transcription_id} - Workflow actors can add a comment against the transcription status
* GET /transcriptions/{transcription_id}/document - Download transcript file

User
* GET /admin/users
* GET /admin/users/{user_id}
* PATCH /admin/users/{user_id} - Modify user by id
* POST /admin/users/search - Get users by search criteria

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/integrationTest/java/uk/gov/hmcts/darts/cases/controller/CaseControllerGetAnnotationsIntTest.java [hunk 1]
```diff
@@ -170,6 +175,35 @@ void givenNonSuperAdminJudgeUserReturnForbidden() throws Exception {
             .andExpect(status().isForbidden())
             .andReturn();
 
+        verify(mockUserIdentity).getUserAccount();
+        verify(mockUserIdentity).userHasGlobalAccess(Set.of(JUDGE, SUPER_ADMIN));
+        verifyNoMoreInteractions(mockUserIdentity);
+    }
+
+    @Test
+    void givenSuperUserReturnForbidden() throws Exception {
```
Comment (karen-hedges) [inline src/integrationTest/java/uk/gov/hmcts/darts/cases/controller/CaseControllerGetAnnotationsIntTest.java:184]: This should probably include forbidden for judges annotations

Comment (richardbull) [reply src/integrationTest/java/uk/gov/hmcts/darts/cases/controller/CaseControllerGetAnnotationsIntTest.java:184]: I don't think it's needed as this integration test class is for the same endpoint 
`GET /cases/{case_id}/annotations` - Admin / Judge view for annotations
I just added this alongside the other test (givenNonSuperAdminJudgeUserReturnForbidden) just to have given a "Super User" identity to verify the response

File: src/main/resources/db/migration/common/V1_284__update_security_data_super_user.sql [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+VALUES (nextval('grp_seq'), 12, 'SUPER_USER', true, true, false, 'Super User', CURRENT_TIMESTAMP, 0, CURRENT_TIMESTAMP, 0);
+
+INSERT INTO security_group_user_account_ae (usr_id, grp_id)
+SELECT usr.usr_id, grp.grp_id
+FROM user_account usr,
+security_group grp
+WHERE user_name = 'darts_global_test_user'
+AND grp.group_name = 'SUPER_USER';
+
+INSERT INTO security_group_user_account_ae (usr_id, grp_id)
```
Comment (hemantasharma1129) [inline src/main/resources/db/migration/common/V1_284__update_security_data_super_user.sql:29]: I dont think we have to be dependent on SUPER_ADMIN role at all for SUPER_USER.

File: src/integrationTest/java/uk/gov/hmcts/darts/testutils/stubs/UserAccountStub.java [hunk 3]
```diff
@@ -280,4 +278,18 @@ public UserAccountEntity createSuperAdminUser() {
         return userAccountRepository.saveAndFlush(user);
     }
 
+    @Transactional
```
[the remark this answers is not in the document]
Comment (richardbull) [reply src/integrationTest/java/uk/gov/hmcts/darts/testutils/stubs/UserAccountStub.java:281]: I don't think it applies here @karen-hedges  - this isn't the test class, and I don't think we have any other services to set this up for us in our application.
The `UserAccountStub#createSuperUser` were given here as testutils classes to persist some given setup data that we need to participate in our test transaction that sits outside the application scope transaction we define in our service layer.

File: src/main/resources/db/migration/common/V1_284__update_security_data_super_user.sql (hunk 2 above)
Comment (richardbull) [reply src/main/resources/db/migration/common/V1_284__update_security_data_super_user.sql:29]: I added this here yesterday as I appeared to be missing the `security_group_user_account_ae` link for SUPER_ADMIN for my `darts_global_test_user` in local postgres.
This INSERT will only add the association if it is missing like in my case as I understand that this user exists for devs / testers to use as a generic global test user.  So it felt like the expectation was to have the group link in place.

**Review by hemantasharma1129 — APPROVED** [review]

**Pull request merged**